Tiled displays tilesets wrongly with qt 6.9

**Describe the bug**

See title. thorbjorn confirmed on IRC this is apparently related to qt 6.9.
A solution would be using the appimage or flatpak for now apparently.

**To Reproduce**

1. Use qt 6.9
2. Run tiled.
3. Open any tileset. It is worse with lower zoom levels and with a higher zoom level, the tiles overlap. These tiles are 16x16.
4. See below.

![Image](https://github.com/user-attachments/assets/3ffb664c-f8ab-48fa-916b-1dd0216bcd00)

**Expected behavior**

It would display them as usual.

**Specifications:**
 - OS: Arch Linux
 - Tiled Version: 1.11.2-1

@bjorn Good hunch! Aside from Qt6, I still happen to have an old Qt5 installed, and that's what qbs chose for the build. Sorry for the confusion.

I tried to build again, this time against Qt 6.9.0, but it fails and I get this:
```
[…]/src/libtiled/qtcompat_p.h:31:7: error: conflicting declaration ‘using QStringRef = class QStringView’
[…]
/usr/include/qt6/QtCore/qstringfwd.h:39:7: note: previous declaration as ‘class QStringRef’
   39 | class QStringRef; // defined in qt5compat
```

--- a/NEWS.md
+++ b/NEWS.md
@@ -1,3 +1,7 @@
+### Unreleased
+
+* Workaround tileset view layout regression in Qt 6.9
+
 ### Tiled 1.11.2 (28 Jan 2025)
 
 * YY plugin: Fixed compatibility with GameMaker 2024 (#4132)

--- a/src/tiled/tilesetview.cpp
+++ b/src/tiled/tilesetview.cpp
@@ -95,7 +95,8 @@ public:
                    const QModelIndex &index) const override;
 
 private:
-    void drawFilmStrip(QPainter *painter, QRect targetRect) const;
+    static void drawFilmStrip(QPainter *painter, QRect targetRect);
+
     void drawWangOverlay(QPainter *painter,
                          const Tile *tile,
                          QRect targetRect,
@@ -210,7 +211,7 @@ QSize TileDelegate::sizeHint(const QStyleOptionViewItem & /* option */,
     return QSize(extra, extra);
 }
 
-void TileDelegate::drawFilmStrip(QPainter *painter, QRect targetRect) const
+void TileDelegate::drawFilmStrip(QPainter *painter, QRect targetRect)
 {
     painter->save();
 
@@ -300,6 +301,15 @@ TilesetView::TilesetView(QWidget *parent)
     vHeader->setSectionResizeMode(QHeaderView::ResizeToContents);
     hHeader->setMinimumSectionSize(1);
     vHeader->setMinimumSectionSize(1);
+#if QT_VERSION >= QT_VERSION_CHECK(6, 9, 0)
+    // Since Qt 6.9 we force header mode to "FlexibleWithSectionMemoryUsage",
+    // to avoid what appears to be a bug in that implementation regarding
+    // resizing of sections (https://github.com/mapeditor/tiled/issues/4191).
+    hHeader->setStretchLastSection(true);
+    vHeader->setStretchLastSection(true);
+    hHeader->setStretchLastSection(false);
+    vHeader->setStretchLastSection(false);
+#endif
 
     // Hardcode this view on 'left to right' since it doesn't work properly
     // for 'right to left' languages.

